### ISSUE TITLE
Facet ranges behave differently from docs

### ISSUE THREAD

**Issue by felixh10r** [typesense/typesense#1561]
## Description

According to the [docs](https://typesense.org/docs/0.25.2/api/search.html#facet-ranges), _range start values are inclusive, but the range end values are exclusive in nature._ However, at implementation, the index is showing an obvious off-by-one error.

## Steps to reproduce
Let the data be 
```json
[
  {"value": 0},
  {"value": 1},
  {"value": 2},
  {"value": 3}
]
```

Let's say we want to have two facet values: 
- values that are 0 
- everything above 

## Expected Behavior

Reading the docs, one would expect the `facet_by` parameter to equal `value(isZero:[0, 1], isAbove:[1, ])`, i.e. that the start values are inclusive and the end values are exclusive.

## Actual Behavior

However, with the above parameter, "1" is grouped with `isZero`, not `isAbove`. The only way to get this working as expected is by specifying `value(isZero:[-1, 0], isAbove:[0, ])` (or without a lower bound `value(isZero:[, 0], isAbove:[0, ])`), which is not only quite counter-intuitive, but also the opposite from what is specified in the docs: The range start value is being treated as exclusive and the end value is inclusive.

## Metadata

**Typesense Version**:
1.7.2 (js sdk)
Typesense Cloud (0.25.2)

**OS**:
MacOS

**Comment by kishorenc** [comment]
This is a bug. We have a fix, and will share a patched build in a day or two.

**Comment by kishorenc** [comment]
Fix available in `0.26.0.rc65`
